Sming/sming.mk

```diff
@@ -0,0 +1,662 @@
+#   sming.mk
+#
+#   to do:
+#       - LTO?
+#
+#   done:
+#       - roms with and without bootloader
+#       - automatic flash address caclulation
+#       - automatic linker script creation
+#       - dependency tracking for spiffs sources
+#       - dependency tracking for all header files included in cpp and c sources
+#       - memanalyzer added
+#       - automatic patch of esp_init_data_default.bin for system_get_vdd33()
+#       - improved terminal handling for windows
+#       - removed vpath to avoid source file collisions
+#       - EXTRA_SRC option added
+#       - more specific include paths to avoid header file collisions
+#       - use Python for address calculations if available instead of Perl
+#       - use sed for linker file creation instead of Perl
+#
+#   remarks:
+#       - old variable names kept for compatibility, even if some of them seem
+#         a bit inconsistent (e. g. SPIFF_SIZE, SPIFF_FILES)
+#
+#==============================================================================
+#   user settings
+#------------------------------------------------------------------------------
+#   your flash memory size (required)
+ifeq ("$(SPI_SIZE)","")
+    $(error please set SPI_SIZE)
+endif
+
+#   enable rboot (default: 0)
+RBOOT_ENABLED   ?= 0
+
+#   your source directories (default: app)
+MODULES         ?= app
+
+#   your extra include directies (default: none)
+EXTRA_INCDIR    ?=
+
+#   your extra libraries (default: none)
+EXTRA_LIBS      ?=
+
+#   your extra source files (default: none)
+EXTRA_SRC       ?=
+
+#   your spiffs size (default: 0)
+SPIFF_SIZE      ?= 0
+
+#   your source directory for spiffs files (default: files)
+SPIFF_FILES     ?= files
+
+#   your reserved flash area for user data (default: 0)
+USER_FLASH_SIZE ?= 0
+
+#   patch init data for system_get_vdd33 (default: 0)
+GET_VDD33       ?= 0
+
+#==============================================================================
+#   load os specific settings
+#------------------------------------------------------------------------------
+ifeq ($(OS),Windows_NT)
+	include $(SMING_HOME)/Windows.mk
+else
+	UNAME := $(shell uname -s)
+	include $(SMING_HOME)/$(UNAME).mk
+endif
+
+#==============================================================================
+#   default serial settings
+#------------------------------------------------------------------------------
+#   use the maximum speed supported by your board to get things done faster
+COM_SPEED_ESPTOOL   ?= 921600
```

Comment:
Use the maximum supported speed from most devices - 115200. Not only from some (http://digital.ni.com/public.nsf/allkb/D37754FFA24F7C3F86256706005B9BE7). change the COM_SPEED_SERIAL too. Otherwise we will be overwhelmed with issues saying that flashing aborted, etc.